llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -1379,7 +1379,15 @@ void CallsiteContextGraph<DerivedCCG, FuncTy, CallTy>::
   DenseSet<uint32_t> LastNodeContextIds = LastNode->getContextIds();
   assert(!LastNodeContextIds.empty());
 
+#ifndef NDEBUG
```

Comment:
Here keeping `CreatedNode` and `PrevIterCreatedNode` around in the optimized build should have little impact. Can we  avoid the ifndef (and differing loop header) here and L1461-L1463?

Reply:
yeah, I was concerned that I would get compiler warnings about unused variables, since PrevIterCreatedNode is only read by an assert, but I didn't seem to.